src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -688,21 +688,26 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		this._language = language.toLowerCase();
 	}
 
-	public changeKernel(displayName: string): void {
+	public changeKernel(displayName: string, kernelAlias?: string[]): void {
```

Comment:
since we're passing around an array of kernel aliases in multiple places, the notebook model should save all kernel aliases thaat have been registered instead of passing it around everywhere. The aliases should be exposed as a property that can be accessed from elsewhere.

Reply:
So maybe there's some history here I'm missing - but why is the list of kernels even a part of the Notebook model? i.e. why does every Notebook need to keep its own list of valid kernels? It seems like that should all be accessed through a service (like providers in core) which then contain the necessary data....

And more specifically for this PR - wouldn't aliases' then just be a property of a kernel? Like each kernel should just keep track of any registered aliases for it that are then displayed separately in the UI.

Reply:
I was able to delete all instances where the array is, as I was able to reference the private property instead.